iOS 17.1 beta 2: ManagedMediaSource video playback stops when changing timestampOffset

jyavenard opened the pull request:
#### 3acdede1a2e206a002bf8436467c51296fe223e7
<pre>
iOS 17.1 beta 2: ManagedMediaSource video playback stops when changing timestampOffset
<a href="https://bugs.webkit.org/show_bug.cgi?id=262886">https://bugs.webkit.org/show_bug.cgi?id=262886</a>
rdar://116707511

Reviewed by Eric Carlson.

We set MediaPlayerPrivateMediaSourceAVFObjC::m_playing to false if the video
playback stalled for whatever reason: a condition detected when the effective
playback rate became 0 and a notification was sent.
However, once we finished seeking or once a decoded data is available again
we would resume playback only if m_playback is true.
We need to resume playback if we used to play the video (that is play() was called
and the video wasn&apos;t explictly paused).

The change effectively revert changes made in 266665@main and 266675@main.

Adding test to catch future regressions.

* LayoutTests/media/media-source/media-managedmse-resume-after-stall-expected.txt: Added.
* LayoutTests/media/media-source/media-managedmse-resume-after-stall.html: Added.
* LayoutTests/platform/ios-16/TestExpectations:
* LayoutTests/platform/ipad/TestExpectations:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::MediaPlayerPrivateMediaSourceAVFObjC):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::playInternal):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::pauseInternal):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::shouldBePlaying const):

Canonical link: <a href="https://commits.webkit.org/269290@main">https://commits.webkit.org/269290@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5e3dbe0f41e01b4982d657be52ecf07e66f60f05

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22119 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/22327 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23188 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24011 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20476 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22364 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26578 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22644 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22349 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/22031 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19191 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24861 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/19114 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20045 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26300 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20131 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20270 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24159 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20702 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17623 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/20056 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/19854 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5267 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/24264 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20650 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson reviewed (approved):
r=me

Please update the commit message with the correct regression point before landing

jyavenard committed:
iOS 17.1 beta 2: ManagedMediaSource video playback stops when changing timestampOffset
https://bugs.webkit.org/show_bug.cgi?id=262886
rdar://116707511

Reviewed by Eric Carlson.

We set MediaPlayerPrivateMediaSourceAVFObjC::m_playing to false if the video
playback stalled for whatever reason: a condition detected when the effective
playback rate became 0 and a notification was sent.
However, once we finished seeking or once a decoded data is available again
we would resume playback only if m_playback is true.
We need to resume playback if we used to play the video (that is play() was called
and the video wasn't explictly paused).

The change effectively revert changes made in 266665@main and 266675@main.

Adding test to catch future regressions.

* LayoutTests/media/media-source/media-managedmse-resume-after-stall-expected.txt: Added.
* LayoutTests/media/media-source/media-managedmse-resume-after-stall.html: Added.
* LayoutTests/platform/ios-16/TestExpectations:
* LayoutTests/platform/ipad/TestExpectations:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::MediaPlayerPrivateMediaSourceAVFObjC):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::playInternal):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::pauseInternal):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::shouldBePlaying const):

Canonical link: https://commits.webkit.org/269290@main
